raft/log.go

```diff
@@ -1021,7 +1026,10 @@ func (l *Log) candidateLoop(closing <-chan struct{}) State {
 			l.lock()
 			l.mustSetTermIfHigher(hb.term)
 			if hb.term >= l.term {
+				l.tracef("candidateLoop: new leader: old=%d new=%d", l.leaderID, hb.leaderID)
 				l.leaderID = hb.leaderID
+				l.unlock()
+				return Follower
```

Comment:
@benbjohnson -- yes, that looks correct. There should be no need to return `follower` here, since the `case` statement below should be triggered by the signal sent by `mustSetTermIfHigher`.